compute_endpoint/globus_compute_endpoint/endpoint/endpoint_manager.py

```diff
@@ -165,14 +165,14 @@ def __init__(
                 exit(os.EX_TEMPFAIL)
 
         upstream_ep_uuid = reg_info.get("endpoint_id")
-        if upstream_ep_uuid != endpoint_uuid:
+        if endpoint_uuid and upstream_ep_uuid != endpoint_uuid:
             log.error(
                 "Unexpected response from server: mismatched endpoint id."
                 f"\n  Expected: {endpoint_uuid}, received: {upstream_ep_uuid}"
             )
             exit(os.EX_SOFTWARE)
 
-        self._endpoint_uuid_str = upstream_ep_uuid
+        self._endpoint_uuid_str = str(upstream_ep_uuid)
```

Comment:
Is this redundant?  We've received a JSON document from upstream, and aren't applying a conversion to it, are we?

Reply:
`upstream_ep_uuid` is grabbed from `reg_info`, which could be defined via `__init__` args.